src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -37,6 +37,43 @@ export const configureClient = async (
   }
 };
 
+export const configureTestClient = async (
+  { savedObjects, cryptography }: DataSourceClientParams,
+  dataSource: DataSourceAttributes,
+  openSearchClientPoolSetup: OpenSearchClientPoolSetup,
+  config: DataSourcePluginConfigType,
+  logger: Logger
+): Promise<Client> => {
+  try {
+    const {
+      id,
+      auth: { type, credentials },
+    } = dataSource;
+    let requireDecryption = false;
+
+    const rootClient = getRootClient(dataSource, config, openSearchClientPoolSetup);
+
+    if (type === AuthType.UsernamePasswordType && !credentials?.password && id) {
+      const { attributes: fetchedDataSource } = await getDataSource(id || '', savedObjects);
+      dataSource.auth = {
+        type,
+        credentials: {
+          username: credentials?.username || '',
+          password: fetchedDataSource.auth.credentials?.password || '',
+        },
+      };
+      requireDecryption = true;
+    }
+
+    return getQueryClient(rootClient, dataSource, cryptography, requireDecryption);
+  } catch (error: any) {
+    logger.error(`Failed to get data source client for dataSource: ${dataSource}`);
```

Comment:
Shouldn't this error message be different since this is testing it?

Reply:
This is a message to display that it is not the connection failure. It failed to get the client that makes the http rquests